### DISCUSSION TITLE
Configuring Renovate to recognize nested requirements.txt

### DISCUSSION THREAD

**Discussion by kyleburnett** [renovatebot/renovate#19703] [Request Help] [answered]
I have a repository with python dependencies at the following file paths:
- src/python/examples/1.0.0/requirements.txt
- src/python/examples/1.1.0/requirements.txt
- src/python/examples/2.0.0/requirements.txt

The default fileMatch setting for pip_requirements according to the logs seems to be `(^|/)([\w-]*)requirements\.(txt|pip)$`, but this was not detecting the files above. I figured it might need to be a full match so I've supplied `(^|/)([\\w-./]*)requirements\\.(txt|pip)$` to the `pip_requirements.fileMatch` configuration field, but that still pick up the requirements files like I expected.

Is there anything else I'm missing in order to get Renovate to pick up the Python packages in the requirements.txt files?

# Configuration

<details>
```
{
  "extends": [
    "config:base"
  ],
  "ignoreDeps": ["proknow.azurecr.io/proknow-build"],
  "prConcurrentLimit": 2,
  "packageRules": [
    {
      "matchDepTypes": ["devDependencies"],
      "prPriority": -1
    },
    {
      "matchPackageNames": ["chalk"],
      "allowedVersions": "<5.0.0"
    },
    {
      "matchPackageNames": ["eslint", "depcheck"],
      "groupName": "CI Dependencies",
      "groupSlug": "ci-dependencies",
      "automerge": true,
      "automergeType": "pr",
      "automergeStrategy": "fast-forward",
      "prPriority": 9
    },
    {
      "matchPackageNames": [
        "mocha",
        "mocha-junit-reporter",
        "mocha-multi-reporters",
        "chai",
        "dir-compare"
      ],
      "groupName": "Testing dependencies",
      "groupSlug": "testing",
      "automerge": true,
      "automergeType": "pr",
      "automergeStrategy": "fast-forward",
      "prPriority": 10
    }
  ],
  "pip_requirements": {
    "fileMatch": ["(^|/)([\\w-./]*)requirements\\.(txt|pip)$"]
  },
  "platformAutomerge": true,
  "assignees": ["kyleburnett"],
  "assigneesSampleSize": 1,
  "reviewersFromCodeOwners": true,
  "reviewersSampleSize": 1
}
```
</details>

# Logs

Here are some log messages indicating that Renovate is reading the `pip_requirements.fileMatch` configuration.

```
DEBUG: Using file match: (^|/)([\w-]*)requirements\.(txt|pip)$ for manager pip_requirements
DEBUG: Using file match: (^|/)([\w-./]*)requirements\.(txt|pip)$ for manager pip_requirements
```

**Comment by viceice** [comment, marked as answer]
`config:base` preset has an ignore paths filter, which ignores test and example directories. checkout the docs for buildin presets

**Reply by kyleburnett** [reply]
Thank you! That helped a lot. For future readers' reference: [config:base documentation](https://docs.renovatebot.com/presets-config/#configbase) and [:ignoreModulesAndTests documentation](https://docs.renovatebot.com/presets-default/#ignoremodulesandtests). Since ignorePaths is not mergeable, I was able to add this to my config (notice that `"**/examples/**"` is removed):

```
{
  "ignorePaths": [
    "**/node_modules/**",
    "**/bower_components/**",
    "**/vendor/**",
    "**/__tests__/**",
    "**/test/**",
    "**/tests/**",
    "**/__fixtures__/**"
  ]
}
```

Also worth mentioning that I was able to remove "pip_requirements.fileMatch" from my config.
